### PR TITLE
chore: fix docs on architecture-fault-tolerance.rst

### PR DESCRIPTION
**Pull request by lhvubtqn**

### CODE REVIEW

**Comment by annastuchlik** [comment]
@guy9 Could you have a look at this small update?

File: docs/architecture/architecture-fault-tolerance.rst [hunk 1]
```diff
@@ -51,7 +51,7 @@ The Consistency Level is tunable per operation in CQL.   This is known as :term:
 
 The Consistency Level and Replication Factor both impact performance. The **lower** the Consistency Level and/or Replication Factor, the **faster** the read or write operation.  However, there will be less fault tolerance if a node goes down.
 
-The Consistency Level itself impacts availability. A **higher** Consistency Level (more nodes required to be online)  means less availability with less tolerance to tolerate node failures. A **lower** Consistency Level means more availability and more fault tolerance.
+The Consistency Level itself impacts availability. A **higher** Consistency Level (more nodes required to be online)  means less availability but with higher tolerance to tolerate node failures. A **lower** Consistency Level means more availability but less fault tolerance.
```
Comment (mykaul) [inline docs/architecture/architecture-fault-tolerance.rst:54]: "higher tolerance to tolerate " sounds a bit strange to me.

Comment (lhvubtqn) [reply docs/architecture/architecture-fault-tolerance.rst:54]: > "higher tolerance to tolerate " sounds a bit strange to me.

I think that also. Will rephrase it to "higher fault tolerance".


> "A **lower** Consistency Level means more availability but less fault tolerance." - why is it less fault tolerant? It's not a 1:1 CL=ALL is not very fault tolerant. CL=1 isn't either - if that specific node is down.

This statement follows the previous one: `The Consistency Level and Replication Factor both impact performance. The **lower** the Consistency Level and/or Replication Factor, the **faster** the read or write operation.  However, there will be less fault tolerance if a node goes down.`

Correct me if I am wrong, but what I understand is, in the case of writing data, the **consistency level specifies how many WRITE acks** the coordinator must receive before sending responses to clients. This means that the fewer required acks, the more risky it gets to have data loss (in case all the nodes that send ack are down). 

For example, if CL=1, there will be only one node that confirms the data has been written. So we will suffer from data loss if that node is down.

All in all, the lower the Consistency Level, the less the fault tolerance.

Comment (guy9) [reply docs/architecture/architecture-fault-tolerance.rst]: In case we have CL=1, an operation is successful once it's completed by a single node. That means we have a high fault tolerance. 
If we have CL=ALL, on the other hand, even a single node (replica) failure will cause the operation to fail, meaning we have a low fault tolerance. 
So, as I see it, the lower the CL, the higher the fault tolerance we have.

Comment (lhvubtqn) [reply docs/architecture/architecture-fault-tolerance.rst]: @guy9 hmm, in the case of CL=ALL we will encounter low availability, not low fault tolerance (cause it takes longer for a WRITE/READ operation to finish - aka. receive response).
> Fault tolerance is the ability of a system to continue functioning correctly even in the presence of hardware or software faults or failures. 

> High availability (HA) refers to the ability of a system or application to remain operational and accessible for users even in the face of certain failures. 

In the case of a failure node, the system can still function correctly according to its configuration (which is preventing write when there are not enough WRITE acks), so it is not a low fault tolerant system.

Your concern lies more in the availability (whether users can interact with the system despite failures), rather than fault tolerance (the system can maintain its correctness).
